src/bindings/overlay.js

```diff
@@ -168,7 +178,11 @@ function isElementAllowed(element) {
  * @returns {boolean}
  * @private
  */
-function isAttrAllowed(attr, element) {
+function isAttrAllowed(ns, attr, element) {
+  // Does it have a namespace that matches the element's?
+  if (ns !== element.namespaceURI) {
+    return false;
```

Comment:
This looks like an unnecessary factorization to make the code in line 77 _and_ line 143 both work.  In case of line 143 this check will always pass.  In line 77 you could simply filter non-matching attributes earlier.